infrap4d/CMakeLists.txt

```diff
@@ -5,24 +5,22 @@ project(infrap4d VERSION 0.1.0 LANGUAGES C CXX)
 set(STRATUM_TDI_BIN_DIR ${STRATUM_SOURCE_DIR}/stratum/hal/bin/tdi)
 set(STRATUM_BFRT_BIN_DIR ${STRATUM_SOURCE_DIR}/stratum/hal/bin/barefoot)
 
-if(DPDK_TARGET)
-    set(INFRAP4D_MAIN ${STRATUM_TDI_BIN_DIR}/dpdk/main.cc)
-elseif(TOFINO_TARGET)
-    set(INFRAP4D_MAIN ${STRATUM_TDI_BIN_DIR}/tofino/main.cc)
-endif()
+add_executable(infrap4d infrap4d_main.cc)
 
-add_executable(infrap4d
-    ${INFRAP4D_MAIN}
-)
-
-target_include_directories(infrap4d PRIVATE ${STRATUM_SOURCE_DIR})
+target_include_directories(infrap4d PRIVATE 
+    ${STRATUM_SOURCE_DIR}
+    ${KRNLMON_SOURCE_DIR})
 
 target_link_libraries(infrap4d PRIVATE
     -Wl,--whole-archive
     stratum
     -Wl,--no-whole-archive
+    krnlmon
+    pthread
 )
 
+add_dependencies(infrap4d stratum krnlmon)
```

Comment:
I don't believe the stratum or krnlmon dependencies need to be added explicitly.